frontend/pkg/frontend/frontend.go

```diff
@@ -641,3 +644,50 @@ func (f *Frontend) ArmDeploymentPreflight(writer http.ResponseWriter, request *h
 
 	arm.WriteDeploymentPreflightResponse(writer, preflightErrors)
 }
+
+func (f *Frontend) logSubscriptionDifferences(oldSub, newSub *arm.Subscription) {
+	// Compare states
+	if oldSub.State != newSub.State {
+		f.logger.Info(fmt.Sprintf("Subscription state changed from %s to %s", oldSub.State, newSub.State))
+	}
+
+	// Compare tenantId
+	if oldSub.Properties != nil && newSub.Properties != nil {
+		if oldSub.Properties.TenantId != nil && newSub.Properties.TenantId != nil {
+			if *oldSub.Properties.TenantId != *newSub.Properties.TenantId {
+				f.logger.Info(fmt.Sprintf("Subscription tenantId changed from %s to %s", *oldSub.Properties.TenantId, *newSub.Properties.TenantId))
+			}
+		}
+	}
+
+	// Compare registeredFeatures
+	if oldSub.Properties != nil && newSub.Properties != nil {
+		if oldSub.Properties.RegisteredFeatures != nil && newSub.Properties.RegisteredFeatures != nil {
+			oldFeatures := featuresMap(oldSub.Properties.RegisteredFeatures)
+			newFeatures := featuresMap(newSub.Properties.RegisteredFeatures)
+			for featureName, oldState := range oldFeatures {
+				newState, exists := newFeatures[featureName]
+				if !exists {
+					f.logger.Info(fmt.Sprintf("Feature %s removed", featureName))
+				} else if oldState != newState {
+					f.logger.Info(fmt.Sprintf("Feature %s state changed from %s to %s", featureName, oldState, newState))
+				}
+			}
+			for featureName, newState := range newFeatures {
+				if _, exists := oldFeatures[featureName]; !exists {
+					f.logger.Info(fmt.Sprintf("Feature %s added with state %s", featureName, newState))
+				}
+			}
+		}
+	}
+}
+
+func featuresMap(features *[]arm.Feature) map[string]string {
+	featureMap := make(map[string]string)
```

Comment:
I think if we modify a bit the code, we can pass to "make" the capacity of the map (len(*features))as a second argument as it is known beforehand (we should check first that it is not nil, though, as the range now protects against nil cases).